doctr/models/classification/predictor/pytorch.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (C) 2021, Mindee.
+
+# This program is licensed under the Apache License version 2.
+# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0.txt> for full license details.
+
+from typing import List, Union
+
+import numpy as np
+import torch
+from torch import nn
+
+from doctr.models.preprocessor import PreProcessor
+
+__all__ = ['OrientationClassifier']
+
+
+class OrientationClassifier(nn.Module):
+    """Implements an object able to detect the reading direction of a text box.
+    4 possible orientations: 0, 90, 180, 270 degrees counter clockwise.
+
+    Args:
+        pre_processor: transform inputs for easier batched model inference
+        model: core classification architecture (backbone + classification head)
+    """
+
+    def __init__(
+        self,
+        pre_processor: PreProcessor,
+        model: nn.Module,
+    ) -> None:
+
+        super().__init__()
+        self.pre_processor = pre_processor
+        self.model = model.eval()
+
+    @torch.no_grad()
+    def forward(
+        self,
+        crops: List[Union[np.ndarray, torch.Tensor]],
+    ) -> List[int]:
+
+        # Dimension check
+        if any(crop.ndim != 3 for crop in crops):
+            raise ValueError("incorrect input shape: all pages are expected to be multi-channel 2D images.")
+
+        processed_batches = self.pre_processor(crops)
+        predicted_batches = [
+            self.model(batch)
+            for batch in processed_batches
+        ]
+
+        # Postprocess predictions
+        predicted_batches = [out_batch.argmax(dim=1).cpu().detach().numpy() for out_batch in predicted_batches]
+
+        return [int(pred) for batch in predicted_batches for pred in batch]
```

Comment:
Mmmh, should we return the class index (1) or the class value (90)?

Reply:
I think it is better to return the class index, because I made the classes so that np.rot90(tensor, class) and np.roll(points, class) rotates the points the right way to reorder points of the crop (cf the next PR):

```
def rectify_crops(
    crops: List[np.ndarray],
    orientations: List[int],
) -> List[np.ndarray]:
    """Rotate each crop of the list according to the predicted orientation:
    0: already straight, no rotation
    1: 90 ccw, rotate 3 times ccw
    2: 180, rotate 2 times ccw
    3: 270 ccw, rotate 1 time ccw
    """
    # Inverse predictions (if angle of +90 is detected, rotate by -90)
    orientations = [4 - pred for pred in orientations if pred != 0]
    return [np.rot90(crop, orientation) for orientation, crop in zip(orientations, crops)]


def rectify_loc_preds(
    page_loc_preds: np.ndarray,
    orientations: List[int],
) -> np.ndarray:
    """Orient the quadrangle (Polygon4P) according to the predicted orientation,
    so that the points are in this order: top L, top R, bot R, bot L if the crop is readable
    """
    return np.stack(
        [np.roll(
            page_loc_pred,
            orientation,
            axis=0) for orientation, page_loc_pred in zip(orientations, page_loc_preds)],
        axis=0
    )
```

Reply:
(I mean we don't have to do a number of rotation --> angle conversion to redo afterwards an angle --> number of rotation conversion)